### PR TITLE
prov/efa: fix bugs in rnr cq entry and test

### PR DESCRIPTION
**Pull request by wzamazon**
Fix bugs in writting cq error entry for RNR, and update test.

### CODE REVIEW

File: fabtests/prov/efa/src/rdm_rnr_read_cq_error.c [hunk 1]
```diff
@@ -82,6 +83,16 @@ static int rnr_read_cq_error(void)
 					rnr_flag = 1;
 					printf("Got RNR error CQ entry as expected: %d, %s\n",
 						comp_err.err, fi_strerror(comp_err.err));
+					prov_errmsg = fi_cq_strerror(txcq, comp_err.prov_errno,
+								     comp_err.err_data,
+								     comp_err.buf,
+								     comp_err.len);
+					if (strstr(prov_errmsg, "Receiver not ready")==NULL) {
+						printf("Got unexpected provider error message.\n");
+						printf("    Expected error message to have \"Receiver not read\" in it\n");
```
Comment (darrylabbate) [inline fabtests/prov/efa/src/rdm_rnr_read_cq_error.c:92]: Typo?

```diff
- printf("    Expected error message to have \"Receiver not read\" in it\n");
+ printf("    Expected error message to have \"Receiver not ready\" in it\n");
```

Comment (wzamazon) [reply fabtests/prov/efa/src/rdm_rnr_read_cq_error.c]: Thank you! fixed in new revision.

**Comment by wzamazon** [comment]
LANL CI failed due "No space left on device"

Error message:
```
tmp/ccyolydg.s: Fatal error: prov/opx/src/.libs/src_libfabric_la-fi_opx_cq_ops_table_non_locking.o: No space left on device
make[2]: *** [Makefile:22788: prov/opx/src/src_libfabric_la-fi_opx_cq_ops_table_non_locking.lo] Error 1
```

File: prov/efa/src/rxr/rxr_op_entry.h [hunk 2]
```diff
@@ -207,6 +213,33 @@ struct rxr_op_entry {
...
+ *
+ * Each op_entry is for an send or receive operation.
+ * Note all packet entries are associated with an op entry.
+ * For example:
+ *     HANDSHAKE packets are not assoicated any send/receive operation
+ *     RMA_CONTEX packets can be associated with a rxr_read_entry.
+ *
+ * @param		pk_entry[in]		packet entry
+ * @return		pointer to the op_entry if the inpu packet entry is associated with an op
+ * 			NULL otherwise
```
Comment (wenduwan) [inline prov/efa/src/rxr/rxr_op_entry.h:227]: A couple typos:
- '...a send or receive...'
- 'Not all packet entries...'
- 'HANDSHAKE packets are not associated with...'
- 'RMA_CONTEXT packets...'
- '@param[in] pkt_entry'
- '...if the intput packet entry...'

File: prov/efa/src/rxr/rxr_op_entry.h [hunk 3]
```diff
@@ -207,6 +213,33 @@ struct rxr_op_entry {
...
+ */
+static inline
+struct rxr_op_entry *rxr_op_entry_of_pkt_entry(struct rxr_pkt_entry *pkt_entry)
```
Comment (wenduwan) [inline prov/efa/src/rxr/rxr_op_entry.h:230]: Why not name it `rxr_x_entry_of_pkt_entry` since we are extracting `pkt_entry->x_entry`?

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_op_entry.h:230]: `x_entry` can be tx_entry, rx_entry or read_entry.

op_entry is more precise here.

File: prov/efa/src/rxr/rxr_pkt_entry.h [hunk 4]
```diff
@@ -175,21 +175,23 @@ struct rxr_pkt_rx_key {
 	fi_addr_t addr;
 };
 
+struct rxr_op_entry;
```
Comment (wenduwan) [inline prov/efa/src/rxr/rxr_pkt_entry.h:178]: Why is this declaration necessary?

File: fabtests/prov/efa/src/rdm_rnr_read_cq_error.c [hunk 5]
```diff
@@ -82,6 +83,16 @@ static int rnr_read_cq_error(void)
 					rnr_flag = 1;
 					printf("Got RNR error CQ entry as expected: %d, %s\n",
 						comp_err.err, fi_strerror(comp_err.err));
+					prov_errmsg = fi_cq_strerror(txcq, comp_err.prov_errno,
+								     comp_err.err_data,
+								     comp_err.buf,
+								     comp_err.len);
+					if (strstr(prov_errmsg, "Receiver not ready")==NULL) {
```
Comment (wenduwan) [inline fabtests/prov/efa/src/rdm_rnr_read_cq_error.c:90]: Missing space around `==`

File: fabtests/prov/efa/src/rdm_rnr_read_cq_error.c [hunk 6]
```diff
@@ -82,6 +83,16 @@ static int rnr_read_cq_error(void)
...
+						printf("Got unexpected provider error message.\n");
+						printf("    Expected error message to have \"Receiver not ready\" in it\n");
+						printf("    Got: %s\n", prov_errmsg);
+						return FI_EINVAL;
```
Comment (wenduwan) [inline fabtests/prov/efa/src/rdm_rnr_read_cq_error.c:94]: Should return a negative code.

File: prov/efa/src/rxr/rxr_pkt_entry.h (hunk 4 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_entry.h:178]: Note that `rxr_op_entry.h` include `rxr_pkt_entry.h`

but `rxr_pkt_entry.h`still reference `rxr_op_entry` through some pointer types. Therefore this forward declaration is needed.

File: prov/efa/src/rxr/rxr_op_entry.h (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_op_entry.h]: fixed an rephrased. thank you!

File: fabtests/prov/efa/src/rdm_rnr_read_cq_error.c (hunk 6 above)
Comment (wzamazon) [reply fabtests/prov/efa/src/rdm_rnr_read_cq_error.c]: fixed

File: fabtests/prov/efa/src/rdm_rnr_read_cq_error.c (hunk 5 above)
Comment (wzamazon) [reply fabtests/prov/efa/src/rdm_rnr_read_cq_error.c]: fixed.

**Review by wenduwan — APPROVED** [review]

**Pull request merged**